psydac/api/fem.py

```diff
@@ -728,7 +728,7 @@ def allocate_matrices(self, backend=None):
                     trd = trial_degree[j]
                     pads[i,j][:] = np.array([td, trd]).max(axis=0)
         else:
-            pads = test_degree
+            pads = np.maximum(test_degree, trial_degree)
```

Comment:
Apparently the original code was taking the `max` only in the case of a system of equations. See line 728